src/main/java/com/rackspacecloud/metrics/tenantroutingservice/controllers/RoutingController.java

```diff
@@ -34,6 +33,24 @@ public TenantRoutes setTenantRoutingInformation(
         return routingInformation;
     }
 
+    @RequestMapping(
+            value = "/full/{tenantId}",
+            method = RequestMethod.POST,
+            consumes = MediaType.APPLICATION_JSON_VALUE,
+            produces = MediaType.APPLICATION_JSON_VALUE
+    )
+    public TenantRoutes fullCreateTenantRoutingData(
+            @PathVariable final String tenantId,
+            @RequestBody final TenantRoutes ingestionRoutingInformationInput
```

Comment:
When this is created I would like to make sure that certain values are not null. But both the @NotNull notation and the lombok @NonNull don't seem to work. Furthermore changing the @AllArgsConstructor in TenantRoutes to a @RequiredArgsConstructor also doesn't generate a nullPointerException like I would have expected. 

It might be a non-issue right now but it is something that should probably be fixed.

Reply:
Ahhha! I knew it had to be some sort of magic invocation that I was missing. 

So with making sure that objects are created with certain parameters do you just recommend using the validation you recommended and the @AllArgsConstructor?

Reply:
Correct, if it's truly for REST API model validation, then keep the class a simple `@Data` class, don't use any of the `@*Constructor` lombok annotations, and let Spring Validation (with `@Validated` :) ) ensure the API caller filled all the required properties.